src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -30,13 +30,21 @@ internal class MsalServiceExceptionFactory
 
             if (IsInvalidGrant(oAuth2Response?.Error, oAuth2Response?.SubError) || IsInteractionRequired(oAuth2Response?.Error))
             {
+                var errorMessageToUse = errorMessage;
+
                 if (IsThrottled(oAuth2Response))
                 {
-                    ex = new MsalUiRequiredException(errorCode, MsalErrorMessage.AadThrottledError, innerException);
+                    errorMessageToUse = MsalErrorMessage.AadThrottledError;
```

Comment:
Set the `errorMessage` and reuse? 
```suggestion
                    errorMessage = MsalErrorMessage.AadThrottledError;
```

Reply:
Why should I do this? errorMessage  is an incoming parameter